cpp/ql/lib/semmle/code/cpp/models/implementations/StructuredExceptionHandling.qll

```diff
@@ -1,9 +1,29 @@
 import semmle.code.cpp.models.interfaces.Throwing
 
-class WindowsDriverFunction extends ThrowingFunction {
-  WindowsDriverFunction() {
+/**
+ * The default behavior for SEH is any function may (conditionally) raise an exception.
+ * NOTE: this can be overriden by for any specific function to make in
+ * unconditional or non-throwing. IR generation will enforce
+ * the most strict interpretation.
+ */
+class DefaultSEHExceptionBehavior extends ThrowingFunction {
+  DefaultSEHExceptionBehavior() { this = any(Function f) }
```

Comment:
I don't think we need this charpred as `ThrowingFunction` is already a `Function` (which Code Scanning is also telling us 🎉)

Reply:
Let me ask this. It is currently defined as a function yes, but in my refactor of this concept I had intended that an ExceptionAnnotation can be on any expression. Hence, this char pred exists for that concept, but you are right it is not needed given I didn't think I could swing that concept for this pr. The question is, given I would hope it could generalize, should the char pred be defensive?

Reply:
I'm fine with leaving it here. Although, I don't think we'll ever generalize `ThrowingFunction` to ever not be a function since that would probably be a bit confusing. Rather, if we generalize it to `ThrowingExpr` I think we'd keep `ThrowingFunction` as a special case that extends both `Function` and `ThrowingExpr`.

Reply:
I went with this 'any' approach, but depending on your comments I can revert it.